diff --git a/client/pthread_create_linux.cc b/client/pthread_create_linux.cc
index 86ff8dba..9608c4aa 100644
--- a/client/pthread_create_linux.cc
+++ b/client/pthread_create_linux.cc
@@ -1,3 +1,5 @@
+// client/pthread_create_linux.cc with musl support
+//
 // Copyright 2020 The Crashpad Authors
 //
 // Licensed under the Apache License, Version 2.0 (the "License");
@@ -43,6 +45,13 @@ void* InitializeSignalStackAndStart(StartParams* params) {
 
 extern "C" {
 
+#ifdef MUSL
+int __pthread_create(pthread_t* res,
+                     const pthread_attr_t* attrp,
+                     void* (*entry)(void*),
+                     void* arg);
+#endif
+
 __attribute__((visibility("default"))) int pthread_create(
     pthread_t* thread,
     const pthread_attr_t* attr,
@@ -50,6 +59,9 @@ __attribute__((visibility("default"))) int pthread_create(
     void* arg) {
   static const crashpad::NoCfiIcall<decltype(pthread_create)*>
       next_pthread_create([]() {
+#ifdef MUSL
+        return (void*)__pthread_create;
+#endif
         const auto next_pthread_create = dlsym(RTLD_NEXT, "pthread_create");
         CHECK(next_pthread_create) << "dlsym: " << dlerror();
         return next_pthread_create;
diff --git a/handler/win/wer/crashpad_wer.cc b/handler/win/wer/crashpad_wer.cc
index 1394e203..441cffbb 100644
--- a/handler/win/wer/crashpad_wer.cc
+++ b/handler/win/wer/crashpad_wer.cc
@@ -20,9 +20,11 @@
 #include "util/misc/address_types.h"
 #include "util/win/registration_protocol_win_structs.h"
 
-#include <Windows.h>
+#include <windows.h>
 #include <werapi.h>
 
+#include <cstddef>
+
 namespace crashpad::wer {
 namespace {
 using crashpad::WerRegistration;
diff --git a/handler/win/wer/crashpad_wer.h b/handler/win/wer/crashpad_wer.h
index ee99e40c..270c3a90 100644
--- a/handler/win/wer/crashpad_wer.h
+++ b/handler/win/wer/crashpad_wer.h
@@ -15,7 +15,7 @@
 #ifndef CRASHPAD_HANDLER_WIN_WER_CRASHPAD_WER_H_
 #define CRASHPAD_HANDLER_WIN_WER_CRASHPAD_WER_H_
 
-#include <Windows.h>
+#include <windows.h>
 #include <werapi.h>
 
 namespace crashpad::wer {
diff --git a/handler/win/wer/crashpad_wer_main.cc b/handler/win/wer/crashpad_wer_main.cc
index 1b767e38..1af7617c 100644
--- a/handler/win/wer/crashpad_wer_main.cc
+++ b/handler/win/wer/crashpad_wer_main.cc
@@ -17,7 +17,7 @@
 
 #include "handler/win/wer/crashpad_wer.h"
 
-#include <Windows.h>
+#include <windows.h>
 #include <werapi.h>
 
 // Functions that will be exported from the DLL.
diff --git a/util/net/http_transport_libcurl.cc b/util/net/http_transport_libcurl.cc
index c8d34644..abeec83d 100644
--- a/util/net/http_transport_libcurl.cc
+++ b/util/net/http_transport_libcurl.cc
@@ -100,6 +100,7 @@ class Libcurl {
   }
 
   bool Initialize() {
+#ifndef MUSL
     void* libcurl = []() {
       std::vector<std::string> errors;
       for (const auto& lib : {
@@ -122,7 +123,18 @@ class Libcurl {
     if (!libcurl) {
       return false;
     }
+#endif
 
+#ifdef MUSL
+#define LINK_OR_RETURN_FALSE(symbol)       \
+  do {                                     \
+    symbol##_.SetPointer(symbol);          \
+    if (!symbol##_) {                      \
+      LOG(ERROR) << "dlsym:" << dlerror(); \
+      return false;                        \
+    }                                      \
+  } while (0);
+#else
 #define LINK_OR_RETURN_FALSE(symbol)               \
   do {                                             \
     symbol##_.SetPointer(dlsym(libcurl, #symbol)); \
@@ -131,6 +143,7 @@ class Libcurl {
       return false;                                \
     }                                              \
   } while (0);
+#endif
 
     LINK_OR_RETURN_FALSE(curl_easy_cleanup);
     LINK_OR_RETURN_FALSE(curl_easy_init);
